tests/pkg/fault-trigger/manager/manager.go

```diff
@@ -15,36 +15,26 @@ package manager
 
 import (
 	"fmt"
-	"sync"
+
+	"github.com/pingcap/tidb-operator/tests/slack"
 )
 
 // Manager to manager fault trigger
 type Manager struct {
-	sync.RWMutex
-	vmCache map[string]string
+	VMManager
 }
 
 // NewManager returns a manager instance
-func NewManager() *Manager {
-	return &Manager{
-		vmCache: make(map[string]string),
+func NewManager(vmManagerName string) *Manager {
+	var vmManager VMManager
+	if vmManagerName == "qm" {
+		vmManager = &QMVMManager{}
+	} else if vmManagerName == "virsh" {
+		vmManager = &VirshVMManager{}
+	} else {
+		slack.NotifyAndPanic(fmt.Errorf("stability test have not supported the vm manager:[%s],please choose [qm] or [virsh].", vmManagerName))
```

Comment:
in fault-trigger, slack.Webhook endpoint is not provided. IMO a simple fatal error is enough.